flipperdevices/flipperzero-firmware
# [FL-3025] IR button overflow fix

# What's new

- Fix #2096 
- Allow for more than 253 buttons in an Infrared remote
- Refactor `button_menu` and `submenu` view code

# Verification 

1. Save `IR-260.ir` from the above mentioned issue to `/ext/infrared`.
2. Go to `Infrared -> Saved Remotes` and select `IR-260`
3. Scroll through the list (**hint:** press `UP` to go straight to the bottom of the list`). No graphical glitches should occur.
4. Press `Edit -> Rename Button` and ensure that the list of button names is also displayed without glitches.

# Checklist (For Reviewer)
